source/components/JsxParser.tsx

```diff
@@ -141,11 +140,13 @@ export default class JsxParser extends React.Component<TProps> {
 			case 'ExpressionStatement':
 				return this.#parseExpression(expression.expression, scope)
 			case 'Identifier':
+				if (expression.name === 'Infinity') return Infinity
+				if (expression.name === '-Infinity') return -Infinity
+				if (expression.name === 'NaN') return NaN
```

Comment:
`Infinity`, `-Infinity`, and `NaN` are valid JS values — but we were throwing them away and incorrectly using them prior.